scripts/src/java/org/oppia/android/scripts/coverage/CoverageReporter.kt

```diff
@@ -0,0 +1,310 @@
+package org.oppia.android.scripts.coverage
+
+import org.oppia.android.scripts.proto.CoverageReport
+import org.oppia.android.scripts.proto.Coverage
+import java.io.File
+
+class CoverageReporter(
```

Comment:
Some thoughts (without reviewing the code in significant detail on this pass):
- As I was mentioning in the 1.4 PR, the expectation is that MD/HTML report generation is for the total summary of results from an entire run of coverage, not for a single target.
- This might differ from a comment I left below about the extension ending in HTML, but I was actually expecting that the HTML file generated would be a rich annotation of the source file being covered (e.g. with red/green/non highlighted lines). This conflicts a bit with a comment I left below about detecting the report type based on extension, but it's adaptable: HTML would require the user to pass a directory path wherein we would probably replicate the existing source file structure and generate an HTML file for each source file covered by tested targets.

I think that you need to think about code coverage, as a whole, like a pipeline:
- We can run coverage for a single target and get a single report for that target.
- We can run coverage for a bunch of targets by running the coverage script for each one in isolation (leading to a bunch of code coverage reports).
- We can combine all of these reports into a single, combined/aggregated report.
- We can generate either a single summarization of all the results (MD), or a rich view of exact coverage on a per-source file basis (HTML) using this combined/aggregated report.

I actually thought the original plan in the design document was to introduce multiple scripts, but perhaps I'm misremembering whether part of that is planned for M2. I can see how the existing technical requirements written in the issue don't go into significant detail on the multi vs single target situation (I intended for M1 to be multiple targets only hence why M2 had a deliverable for single target support).

I do recall we discussed this in some detail in the design document, and I had thought the document moved ultimately toward this approach. It'd be really helpful to understand what you're thinking here broadly.

Reply:
Follow-up: ah I think part of my comment above is incorrect since I just realized you are, indeed, coloring the lines of the source file in the generated HTML (it looks really nice!), but the single vs. multi targets part is still the key difference, I think, that needs to be figured out.

Reply:
As discussed, we will need to do / add 2 things but for M2,
1. Make sure multi targets work in a cumulative way for a single source file. ie. if app/main/java/.../HomeActivity.kt is provided as developer/user input -> this should fetch coverage report from both HomeActivityLocalTest and HomeActivityTest targets for the single source file and present them as a combined report.
2. We need to have the same way of executing ci checks for local checking too. ie. as ci runs will need be based on multiple files/targets provided, those need to be doable in local development too. So something like
```
bazel run //scripts:run_coverage -- $(pwd) [utility/.../MathModel.kt, app/.../HomeActivity.kt, app/..../HelpActivity.kt]
``` 
should be doable

Reply:
Verifying that for this PR, the script runs for a single source file.